### PR TITLE
feat: add target filter

### LINKED ISSUE #8: LogLevelModuleFilter
Something like fern's `level_for` - https://docs.rs/fern/latest/fern/struct.Dispatch.html#method.level_for

It should be easy to implement by extending the code of `LogLevelFilter` with a module field.

### PR DESCRIPTION
**Pull request by andylokandy**
closes #8

### CODE REVIEW

**Comment by tisonkun** [comment]
@andylokandy this should close #8

I have some questions when considering the design of this filter, you can take a look https://github.com/fast/logforth/issues/8#issuecomment-2265094826

Those can be faq when using this filter.

**Comment by andylokandy** [comment]
Yes. I'm following the idea in #8. 

Q: If multiple LogLevelModuleFilter configured? What's the manner? Especially if the targets are overlapped
A: Simply apply by the filter order

Q: What is the priority between LogLevelModuleFilter and LevelFilter.
A: LogLevelModuleFilter is super set of LevelFilter. LevelFilter is simpler to use.

Q: Filters are applied in order, but when to short circuit the test (by Accepted/Rejected)
A: LevelFilter will reject if level is lower than levelfilter, and LogLevelModuleFilter will reject if level is lower than levelfilter and the target is matched

File: src/filter/level.rs [hunk 1]
```diff
@@ -17,7 +17,7 @@ use log::Metadata;
 use crate::filter::Filter;
 use crate::filter::FilterResult;
 
-/// A filter that checks if the log level is at most the specified level.
+/// A filter that checks if the log level is at higher than the specified level.
```
Comment (tisonkun) [inline src/filter/level.rs:20]:
```suggestion
/// A filter that checks if the log level is higher than the specified level.
```

**Review by tisonkun — APPROVED** [review]
LGTM. One minor comment inline.

File: src/filter/target.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+use log::Metadata;
+
+use crate::filter::Filter;
+use crate::filter::FilterResult;
+
+/// A filter that checks if the log level is higher than the specified level for a specific
+/// target.
+///
+/// If the target has a prefix that matches the target of the log record, the filter will be
```
Comment (tisonkun) [inline src/filter/target.rs:25]:
```suggestion
/// Only if the target has a prefix that matches the target of the log record, the filter will be
```

File: src/filter/target.rs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+/// applied.
```
Comment (tisonkun) [inline src/filter/target.rs:26]:
```suggestion
/// Only if the target has a prefix that matches the target of the log record, the filter
/// will be applied.
```

**Pull request merged**